enclave/build.rs

```diff
@@ -0,0 +1,12 @@
+use std::path::Path;
+
+use libbpf_cargo::SkeletonBuilder;
+
+fn main() {
+    let bpf_dir = Path::new("src").join("bpf");
+    let src = bpf_dir.join("enclave.bpf.c");
+    let skel = bpf_dir.join("mod.rs");
+
+    SkeletonBuilder::new(&src).generate(&skel).unwrap();
+    println!("cargo:rerun-if-changed={}", src.to_str().unwrap());
+}
```

Comment:
You could do the following:

```rust
fn main() -> Result<(), Box<dyn std::errorError>> {
    let bpf_dir = Path::new("src").join("bpf");
    let src = bpf_dir.join("enclave.bpf.c");
    let skel = bpf_dir.join("mod.rs");

    SkeletonBuilder::new(&src).generate(&skel)?);
    println!("cargo:rerun-if-changed={}", src.to_str()?);
}

Reply:
After doing so, I got

```
error[E0277]: the trait bound `NoneError: std::error::Error` is not satisfied
  --> enclave/build.rs:11:55
   |
11 |     println!("cargo:rerun-if-changed={}", src.to_str()?);
   |                                                       ^ the trait `std::error::Error` is not implemented for `NoneError`
   |
   = note: required because of the requirements on the impl of `From<NoneError>` for `Box<dyn std::error::Error>`
   = note: required by `from`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `enclave`
```

I also see that there is no trait for `NoneError`. I'm tempted to just use `anyhow` here.

Reply:
after adding `anyhow` as build dependency and changing the function siguature to:

```rust
fn main() -> anyhow::Result<()> {
```

I'm apparently getting the same error:

```
error[E0277]: the trait bound `NoneError: std::error::Error` is not satisfied
  --> enclave/build.rs:11:55
   |
11 |     println!("cargo:rerun-if-changed={}", src.to_str()?);
   |                                                       ^ the trait `std::error::Error` is not implemented for `NoneError`
   |
   = note: required because of the requirements on the impl of `From<NoneError>` for `anyhow::Error`
   = note: required by `from`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `enclave`
```

Should I maybe do something like this?

```rust
println!("cargo:rerun-if-changed={}", src.to_str().ok_or(MyStrconvError)?;
```

I have the similar line of code in `main.rs` which works.

Reply:
I did that, PTAL, let me know if you agree ^

Reply:
Looks great to me! Agreed with your analysis and solution.